### PR TITLE
[API] Add product review validation

### PR DESCRIPTION
**Pull request by arti0090**
| Q               | A
| --------------- | -----
| Branch?         | 1.9
| Bug fix?        | yes, kind of
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| License         | MIT

### CODE REVIEW

File: .idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Default ignored files
```
Comment (Tomanhez) [inline .idea/.gitignore:1]: Not needed :)

File: .idea/Sylius.iml [hunk 2]
```diff
@@ -0,0 +1,307 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Tomanhez) [inline .idea/Sylius.iml:1]: And this file too

**Review by Tomanhez — COMMENTED** [review]
You should remove `.idea` files

File: .idea/php-test-framework.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (SirDomin) [inline .idea/php-test-framework.xml:1]: i think your ubuntu doesnt ignore .idea/* files :D

File: src/Sylius/Bundle/ApiBundle/spec/Validator/Constraints/UniqueReviewerEmailValidatorSpec.php [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+        $this->validate(null, new UniqueReviewerEmail());
+    }
+
+    function it_throws_an_exception_if_constraint_is_not_of_expected_type(): void
+    {
+        $this
+            ->shouldThrow(\InvalidArgumentException::class)
+            ->during('validate', ['', new class() extends Constraint {
+            }])
```
Comment (SirDomin) [inline src/Sylius/Bundle/ApiBundle/spec/Validator/Constraints/UniqueReviewerEmailValidatorSpec.php:69]:
```suggestion
            ->during('validate', ['', new class() extends Constraint {}])
```
What do you think?

File: src/Sylius/Bundle/ApiBundle/Validator/Constraints/UniqueReviewerEmailValidator.php [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        if ($value === $this->getAuthenticatedUserEmail()) {
+            return;
+        }
+
+        if ($this->shopUserRepository->findOneByEmail($value) !== null) {
+            $this->context->addViolation($constraint->message);
+        }
+    }
+
+    private function getAuthenticatedUserEmail(): ?string
```
Comment (SirDomin) [inline src/Sylius/Bundle/ApiBundle/Validator/Constraints/UniqueReviewerEmailValidator.php:57]: Maybe we could move this private function to new service?

File: .idea/.gitignore (hunk 1 above)
Comment (arti0090) [reply .idea/.gitignore:1]: Lol

File: src/Sylius/Bundle/ApiBundle/spec/Validator/Constraints/UniqueReviewerEmailValidatorSpec.php (hunk 4 above)
Comment (arti0090) [reply src/Sylius/Bundle/ApiBundle/spec/Validator/Constraints/UniqueReviewerEmailValidatorSpec.php:69]: Maybe it is good, but I stayed with original files that we have in branches 1.10^ so maybe upmerge would be easier. Same for your next comment.

**Review by AdamKasp — APPROVED** [review]

**Pull request closed without merging**